integration-tests/smoke/keeper_test.go

```diff
@@ -425,6 +429,84 @@ func getKeeperSuite(
 					}, "1m", "1s").Should(Succeed())
 				})
 			}
+
+			if testToRun == AddFundsToUpkeepTest {
+				It("adds funds to a new underfunded upkeep", func() {
+					listOfNewUpkeeps := actions.DeployKeeperConsumers(contractDeployer, networks, 1)
+					newUpkeep := listOfNewUpkeeps[0]
+					newUpkeepAddress := listOfNewUpkeeps[0].Address()
+
+					req, err := registrar.EncodeRegisterRequest(
+						fmt.Sprintf("upkeep_%d", len(upkeepIDs)),
+						[]byte("0x1234"),
+						newUpkeepAddress,
+						upkeepGasLimit,
+						networks.Default.GetDefaultWallet().Address(),
+						[]byte("0x"),
+						big.NewInt(1),
+						0,
+					)
+					Expect(err).ShouldNot(HaveOccurred(), "Could not encode first register request")
+
+					// We want the new upkeep to be initially underfunded, so just transfer a minuscule amount of LINK
+					tx, err := linkToken.TransferAndCall(registrar.Address(), big.NewInt(1), req)
+					Expect(err).ShouldNot(HaveOccurred(), "Could not transfer small amount of LINK")
+					err = networks.Default.WaitForEvents()
+					Expect(err).ShouldNot(HaveOccurred(), "Failed to wait for events")
+
+					receipt, err := networks.Default.GetTxReceipt(tx.Hash())
+					Expect(err).ShouldNot(HaveOccurred(), "Could not obtain transaction receipt")
+
+					var upkeepID *big.Int
+					for _, rawLog := range receipt.Logs {
+						parsedUpkeepId, err := registry.ParseUpkeepIdFromRegisteredLog(rawLog)
+						if err == nil {
+							upkeepID = parsedUpkeepId
+							break
+						}
+					}
+					Expect(upkeepID).ShouldNot(BeNil(), "Upkeep ID not found after registration")
+					log.Info().Msg("Successfully registered new upkeep with ID " + upkeepID.String())
+
+					// Since the upkeep is currently underfunded, check that it doesn't get executed for a while
+					Consistently(func(g Gomega) {
+						counter, err := newUpkeep.Counter(context.Background())
+						g.Expect(err).ShouldNot(HaveOccurred(), "Calling consumer's counter shouldn't fail")
+						g.Expect(counter.Int64()).Should(Equal(int64(0)),
+							"Expected consumer counter to remain zero, but got %d", counter.Int64())
+					}, "30s", "1s").Should(Succeed())
+
+					// Create a new request for the register where we actually fund the upkeep with proper funds
```

Comment:
instead of a new upkeep, can we fund the existing upkeep?

Reply:
That's how I initially tried to do it, but I was getting a weird RPC error without any debugging information and I couldn't fix it. Would be happy to try together maybe you know what I'm missing in case the current way is too complicated.